presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RemoveEmptyDelete.java

```diff
@@ -55,12 +53,15 @@
 public class RemoveEmptyDelete
         implements Rule<TableFinishNode>
 {
-    private static final Capture<ValuesNode> VALUES = newCapture();
-
     private static final Pattern<TableFinishNode> PATTERN = tableFinish()
             .with(source().matching(exchange()
                     .with(onlySource().matching(delete()
-                            .with(source().matching(values().capturedAs(VALUES)))))));
+                            .with(source().matching(emptyValues()))))));
+
+    private static Pattern<ValuesNode> emptyValues()
```

Comment:
I think you can put this to DSL patterns, I would expect to use it more frequently